contributor-docs/versioning.md

```diff
@@ -0,0 +1,132 @@
+|            | A component is deprecated                                     | `major`     |
+|            | A component is removed                                        | `major`     |
+| Props      | A prop is added to a component                                | `minor`     |
+|            | The type of a prop is made more general                       | `minor`     |
+|            | The type of a prop is made more specific                      | `major`     |
+|            | A prop is deprecated                                          | `minor`     |
+|            | A prop is removed from a component                            | `major`     |
+|            | The element to which additional props are added to is changed | `<todo>`    |
+| Markup     | The DOM node that an `id` corresponds to is changed           | `<todo>`    |
+|            | The DOM node that an `aria-label` corresponds to is changed   | `<todo>`    |
+|            | The `role` of a component is changed                          | `<todo>`    |
+| Styles     | The `position` of the outermost element is updated            | `<todo>`    |
+|            | The `display` property of the outermost element is updated    | `<todo>`    |
+|            | A flex or grid property is updated                            | `<todo>`    |
+|            | A selector is added                                           | `<todo>`    |
+|            | A selector is removed                                         | `<todo>`    |
+|            | The specificity of a selector is raised                       | `<todo>`    |
+|            | The specificity of a selector is lowered                      | `<todo>`    |
+| Behavior   | Interactions are added to a component                         | `<todo>`    |
+|            | Interactions are removed from a component                     | `<todo>`    |
+| TypeScript | A type is added                                               | `minor`     |
+|            | A type is made more general                                   | `minor`     |
+|            | A type is made more specific                                  | `major`     |
+|            | A type is removed                                             | `major`     |
+| Package    | An entrypoint is added to the package                         | `minor`     |
+|            | An entrypoint is removed from a package                       | `major`     |
+
+### Examples
+
+#### A prop is added to a component
+
+semver bump: **minor**
+
+```diff
+type Props = {
+  propA: string;
++  propB: string;
+};
+
+function ExampleComponent({
+  propA,
++  propB,
+}: Props) {
+  return (
+    <>
+      <span>{propA}</span>
++      <span>{propB}</span>
+    </>
+  );
+}
+```
+
+#### An existing prop is deprecated
+
+semver bump: **minor**
+
+```diff
+type Props = {
+  propA: string;
++  /**
++   * @deprecated This prop will be removed in the next major version of
```

Comment:
❤️ 

Wondering if these would need to be mopped up at a later date. Is there a mechanism or process you had in mind to do that already?

Reply:
@rezrah I'd honestly love a comment on the issue for next major that is updated every week or whatever cadence with all these with a `@deprecated` comment 😂 

But generally I feel like at the very least if we keep track of deprecations in something like a next major release issue it can help for tracking what's changing, writing migration docs, etc.

Reply:
Yeah, that's nice. We could use a CI workflow to grep the codebase for `@deprecated` annotations and then auto update the next major issue. Good idea 👍